### ISSUE TITLE
Make typescript-eslint more strict for this repo

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/geometric-optics#437]
This can be done by changing the "eslintConfig" section of package.json to something like this:

```js
  "eslintConfig": {
    "extends": "../chipper/eslint/sim_eslintrc.js",
    "overrides": [
      {
        "files": [
          "**/*.ts"
        ],
        "rules": {
          "@typescript-eslint/adjacent-overload-signatures": "error",
          "@typescript-eslint/ban-ts-comment": "error",
          "@typescript-eslint/ban-types": "error",
          "@typescript-eslint/no-explicit-any": "error",
          "@typescript-eslint/no-empty-function": "error"
        }
      }
    ]
  }
```

**Comment by pixelzoom** [comment]
I enabled most  of the rules shown above.  Notes about the ones that I did not enabled:

`ban-types`:  This flags all occurrences of `{}` with:
> Don't use `{}` as a type. `{}` actually means "any non-nullish value".
... and that's ubiquitous with `type SelfOptions = {}`.

`no-non-null-assertion`: The are situations were it's appropriate (and necessary) to use this operator after checking (via an assertion) that the value is truthy.  Using the `range` or `rangeProperty` of a NumberProperty is a particularly common case.
